storage/json_client.go

```diff
@@ -0,0 +1,251 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"strings"
+
+	"golang.org/x/oauth2/google"
+	"google.golang.org/api/option"
+	"google.golang.org/api/option/internaloption"
+	raw "google.golang.org/api/storage/v1"
+	"google.golang.org/api/transport"
+	htransport "google.golang.org/api/transport/http"
+	iampb "google.golang.org/genproto/googleapis/iam/v1"
+)
+
+// jsonStorageClient is the HTTP-JSON API implementation of the transport-agnostic
+// storageClient interface.
+//
+// Note: This is experimental and should not be used in production yet.
+type jsonStorageClient struct {
+	creds    *google.Credentials
+	hc       *http.Client
+	host     string
```

Comment:
Should be readHost (only used for the reader).

Reply:
Done.